### PR TITLE
M2-4443: Audio player item: "Add Audio" button is not clickable for the 1st time

### PR DESCRIPTION
**Pull request by muhurov**

### CODE REVIEW

File: src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/AudioPlayer/AudioPlayerContent/AudioPlayerContent.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import { AddAudio } from '../AddAudio';
+import { UploadAudio } from '../UploadAudio';
+import { RecordAudio } from '../RecordAudio';
+import { RemoveAudioPopup } from '../RemoveAudioPopup';
+
+export const AudioPlayerContent = ({
+  media,
+  setMedia,
+  name,
+}: {
```
Comment (IhorNesterenko99) [inline src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/AudioPlayer/AudioPlayerContent/AudioPlayerContent.tsx]: Pls move type to AudioPlayerContent.types.ts

File: src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/AudioPlayer/AudioPlayerContent/AudioPlayerContent.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+      </StyledTitleMedium>
+      {url && <MLPlayer media={media} onRemove={() => setRemoveAudioPopupOpened(true)} />}
+      {!url && (
+        <AddAudio
+          onUploadAudio={() => setUploadPopupOpened(true)}
+          onRecordAudio={() => setRecordPopupOpened(true)}
+        />
+      )}
+      {error && (
+        <StyledBodyErrorText sx={{ mt: theme.spacing(2.4) }}>{error?.message}</StyledBodyErrorText>
```
Comment (IhorNesterenko99) [inline src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/AudioPlayer/AudioPlayerContent/AudioPlayerContent.tsx]:
```suggestion
        <StyledBodyErrorText sx={{ mt: theme.spacing(2.4) }}>{error.message}</StyledBodyErrorText>
```

File: src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/AudioPlayer/AudioPlayerContent/AudioPlayerContent.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        {t('audioPlayerDescription')}
+      </StyledTitleMedium>
+      {url && <MLPlayer media={media} onRemove={() => setRemoveAudioPopupOpened(true)} />}
+      {!url && (
+        <AddAudio
+          onUploadAudio={() => setUploadPopupOpened(true)}
+          onRecordAudio={() => setRecordPopupOpened(true)}
+        />
+      )}
+      {error && (
```
Comment (IhorNesterenko99) [inline src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/AudioPlayer/AudioPlayerContent/AudioPlayerContent.tsx]:
```suggestion
      {error?.message && (
```

File: src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/AudioPlayer/AudioPlayer.tsx [hunk 4]
```diff
@@ -1,119 +1,52 @@
...
-    onCloseRecordPopup();
-  };
-
-  const handleRemoveAudio = () => {
-    onClearMedia();
-    onCloseRemoveAudioPopup();
-  };
-
-  const HeaderContent = ({ open }: SharedToggleItemProps) =>
+  const HeaderContent = ({ open, media }: SharedToggleItemProps & { media: MediaType | null }) =>
```
Comment (IhorNesterenko99) [inline src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/AudioPlayer/AudioPlayer.tsx]: maybe also move to it's own component?

Comment (alex-i-e) [reply src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/AudioPlayer/AudioPlayer.tsx]: right, it's better to use react reconciliation independently of parent function render

**Review by alex-i-e — APPROVED** [review]

File: src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/AudioPlayer/AudioPlayerContent/AudioPlayerContent.tsx (hunk 1 above)
Comment (muhurov) [reply src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/AudioPlayer/AudioPlayerContent/AudioPlayerContent.tsx]: done

File: src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/AudioPlayer/AudioPlayerContent/AudioPlayerContent.tsx (hunk 2 above)
Comment (muhurov) [reply src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/AudioPlayer/AudioPlayerContent/AudioPlayerContent.tsx]: done

File: src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/AudioPlayer/AudioPlayerContent/AudioPlayerContent.tsx (hunk 3 above)
Comment (muhurov) [reply src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/AudioPlayer/AudioPlayerContent/AudioPlayerContent.tsx]: done

File: src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/AudioPlayer/AudioPlayer.tsx (hunk 4 above)
Comment (muhurov) [reply src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/AudioPlayer/AudioPlayer.tsx]: done

**Review by IhorNesterenko99 — APPROVED** [review]

**Pull request merged**